### PR TITLE
Fix docker build and add installation docs

### PR DESCRIPTION
**Pull request by kentbull**
The docker build was failing on the `blake3` library for me due to not seeing the Cargo environment in the path. I also had issues downloading the `ordered-set` dependency since it requires Python higher than 3.9.7 so I upgraded the Docker image to 3.10.8.

Prefixing the `pip install` with the `source $HOME/.cargo/env` ensures that the `pip install` command sees the Rust environment that was installed in an earlier new step.

I collected the apt-get instructions into one `RUN` instruction for less layers in the final Docker image.

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -1,11 +1,17 @@
-FROM python:3.9.7-buster
+FROM python:3.10.8-buster
```
Comment (pfeairheller) [inline Dockerfile:1]: Can this be changed to 3.10.4 that matches setup.py?

**Review by pfeairheller — CHANGES_REQUESTED** [review]
Please change the python docker image reference to 3.10.4 and we can merge this in.

File: Dockerfile (hunk 1 above)
Comment (kentbull) [reply Dockerfile]: done

**Comment by kentbull** [comment]
I added one extra commit to order the newest changes last in the Dockerfile to take maximum advantage of Docker layer caching. This allows you to only have to download and build the Rust binaries once and then on each incremental Docker image build only the new Python source has to be mounted.

**Comment by SmithSamuelM** [comment]
So should we update keripy pip install to be python 3.10.8?

**Comment by pfeairheller** [comment]
No, the opposite.  I made Kent change the Dockerfile to 3.10.4

**Comment by SmithSamuelM** [comment]
Since we are releasing a new version of KERIpy on PyPI wouldn’t that be a good time to update the version. In general.

**Comment by pfeairheller** [comment]
Yes, that's a good point.  I will give it a try after my current meeting.

**Pull request merged**